axolotl-web/src/pages/MessageList.vue

```diff
@@ -315,6 +359,81 @@ export default {
     scrollDown() {
       document.getElementById("chat-bottom").scrollIntoView();
     },
+    recordAudio(){
+      var that = this;
+      this.playing =  false;
+       navigator.mediaDevices.getUserMedia({
+          video: false,
+          audio: true
+      }).then(async function() {
+
+          that.recorder = new MicRecorder({
+            bitRate: 128
+          });
+          that.recorder.start().then(() => {
+            // something else
+            that.recording = true;
+          }).catch((e) => {
+            /* eslint-disable no-console */
+            //console.error(e);
+          });
+      });
+    },
+    deleteAudio(){
+      this.stopPlayAudio();
+      this.playing = false;
+      this.recorded = false;
+      this.recorder = null;
+    },
+    playAudio(){
+      this.playing = true;
+      this.voiceNoteElem.play();
+
+    },
+    stopPlayAudio(){
+      this.playing = false;
+      this.voiceNoteElem = document.getElementById("voiceNote");
+      this.voiceNoteElem.pause();
+    },
+    stopRecording(){
+      this.recording = false;
+      this.recorded = true;
+      this.recorder.stop()
+      this.recorder.getMp3().then(([_buffer, blob]) => {
+        this.blobUrl = URL.createObjectURL(blob);
+        this.blobObj = blob;
+        let that = this;
+        setTimeout(function(){
+          that.voiceNoteElem = document.getElementById("voiceNote");
+          that.duration = that.voiceNoteElem.duration;
+        }, 200);
+      })
+
+    },
+    sendVoiceNote(){
+      this.recorded = false;
+      let reader = new FileReader();
+      let voiceNoteElem = document.getElementById("voiceNote");
+      voiceNoteElem.pause()
+      this.playing = false;
+      /* eslint-disable no-unused-vars */
+      this.recorder.getMp3().then(([buffer, _blob]) => {
```

Comment:
Doesn't `.then( () => {` work? `.then( ([,]) => {` at least should.

Reply:
It's an artifact from the first implementation. Now it's clean, thank you!